Python/ceval.c

```diff
@@ -4876,61 +4873,61 @@ get_exception_handler(PyCodeObject *code, int index, int *level, int *handler, i
 
 static int
 initialize_locals(PyThreadState *tstate, PyFrameConstructor *con,
-    PyObject **fastlocals, PyObject *const *args,
-    Py_ssize_t argcount, PyObject *kwnames)
+                  PyObject **fastlocals,
+                  PyObject *const *args, Py_ssize_t argcount,
+                  PyObject *kwnames)
 {
     PyCodeObject *co = (PyCodeObject*)con->fc_code;
-    const Py_ssize_t total_args = co->co_argcount + co->co_kwonlyargcount;
-    PyObject **freevars = fastlocals + co->co_nlocals;
+    const int total_args = co->co_argcount + co->co_kwonlyargcount;
 
     /* Create a dictionary for keyword parameters (**kwags) */
     PyObject *kwdict;
-    Py_ssize_t i;
     if (co->co_flags & CO_VARKEYWORDS) {
         kwdict = PyDict_New();
         if (kwdict == NULL)
             goto fail;
-        i = total_args;
+        int offset = _PyCode_OffsetFromIndex(co, total_args, CO_FAST_LOCAL);
```

Comment:
This is performance critical code and this is going to be slower unless `_PyCode_OffsetFromIndex` is inlined.